theano/tensor/basic.py

```diff
@@ -7138,11 +7138,15 @@ def make_node(self, a, choices):
                 "choose first argument must have an [u]int* dtype. Got %s." % a.dtype
             )
 
+        # Only use make_list if choices have inconsisten shapes
+        # otherwise use as_tensor_variable
+        choice_ndim = None
         if isinstance(choices, (tuple, list, theano.typed_list.TypedListVariable)):
-            choice = theano.typed_list.make_list(choices)
-            choice_ndim = choice.ttype.ndim
-            choice_bcast = choice.ttype.broadcastable
-        else:
+            if np.any([choice.ndim > 1 for choice in choices]):
```

Comment:
If you use the built-in `any`, then you can avoid creating a list object by using a generator directly (i.e. `any(c.ndim > 1 for c in choices)`).

Reply:
I tried to use the built-in function, but for a reason I cannot explain the conditions were not handled correctly, probably due to a strange behaviour of tensor object inside the iterable. I finally return to the `np.any([])` approach.

Reply:
What is the exact error/issue?

Reply:
I just found the problem, it looks like using `any()` actually call another function in the same file [any](https://github.com/pymc-devs/Theano-PyMC/blob/4249a01c886d034abf5a29bb40c2a0d06c56106e/theano/tensor/basic.py#L6610) that overwrite the built-in function, which is not best practice I think.

Reply:
Ha, yeah, we should change that to `any_tt`, or use `builtins.any` explicitly.

Reply:
I fixed this by using `builtins.any`, and I removed the `python_any` declaration l.40 for clarity. I did the same for `python_all` and `python_complex` in the file as I found this approach more straightforward. Let me know what you think.